feat(icons): added `list-compact` icon

juliankellydesign opened the pull request:
<!-- Thank you for contributing! -->

<!-- Insert `closes #issueNumber` here if merging this PR will resolve an existing issue -->

## What is the purpose of this pull request?
- [x] New Icon

### Description
Added new `list-compact` icon.

### Icon use case
<!-- What is the purpose of this icon? For each icon added, please insert at least two real life use cases (the more the better). Text like "it's a car icon" is not accepted. -->
We use this icon in Conductor to show collapsing many messages in the chat view. It references Figma's "collapse layers" action

### Alternative icon designs
<!-- If you have any alternative icon designs, please attach them here. -->

## Icon Design Checklist

### Concept
<!-- All of these requirements must be fulfilled. -->
- [x] I have provided valid use cases for each icon.
- [x] I have not added any a brand or logo icon.
- [x] I have not used any hate symbols.
- [x] I have not included any religious or political imagery.

### Author, credits & license
<!-- Please choose one of the following, and put an "x" next to it. -->
- [x] The icons are solely my own creation.
- [x] The icons were originally created in #<issueNumber> by @juliankellydesign
- [x] I've based them on the following Lucide icons: `list`

### Naming
<!-- All of these requirements must be fulfilled. -->
- [x] I've read and followed the [naming conventions](https://lucide.dev/guide/design/icon-design-guide#naming-conventions)
- [x] I've named icons by what they are rather than their use case.
- [x] I've provided meta JSON files in `icons/[iconName].json`.

### Design
<!-- All of these requirements must be fulfilled. -->
- [x] I've read and followed the [icon design guidelines](https://lucide.dev/guide/design/icon-design-guide)
- [x] I've made sure that the icons look sharp on low DPI displays.
- [x] I've made sure that the icons look consistent with the icon set in size, optical volume and density.
- [x] I've made sure that the icons are visually centered.
- [x] I've correctly optimized all icons to three points of precision.

## Before Submitting
- [x] I've read the [Contribution Guidelines](https://github.com/lucide-icons/lucide/blob/main/CONTRIBUTING.md).
- [x] I've checked if there was an existing PR that solves the same issue.

juliankellydesign commented:
<a title="Open lucide studio" target="_blank" href="https://studio.lucide.dev/edit?value=%3Cpath+d=%22m15+19.5+3-3+3+3%22+/%3E%3Cpath+d=%22m15+4.5+3+3+3-3%22+/%3E%3Cpath+d=%22M3+14.5h8%22+/%3E%3Cpath+d=%22M3+19.5h8%22+/%3E%3Cpath+d=%22M3+4.5h8%22+/%3E%3Cpath+d=%22M3+9.5h8%22+/%3E&name=list-compact"><img alt="icons" width="200px" src="https://lucide.dev/api/gh-icon/PHN2ZwogIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIKICB3aWR0aD0iMjQiCiAgaGVpZ2h0PSIyNCIKICB2aWV3Qm94PSIwIDAgMjQgMjQiCiAgZmlsbD0ibm9uZSIKICBzdHJva2U9ImN1cnJlbnRDb2xvciIKICBzdHJva2Utd2lkdGg9IjIiCiAgc3Ryb2tlLWxpbmVjYXA9InJvdW5kIgogIHN0cm9rZS1saW5lam9pbj0icm91bmQiCj4KICA8cGF0aCBkPSJtMTUgMTkuNSAzLTMgMyAzIiAvPgogIDxwYXRoIGQ9Im0xNSA0LjUgMyAzIDMtMyIgLz4KICA8cGF0aCBkPSJNMyAxNC41aDgiIC8+CiAgPHBhdGggZD0iTTMgMTkuNWg4IiAvPgogIDxwYXRoIGQ9Ik0zIDQuNWg4IiAvPgogIDxwYXRoIGQ9Ik0zIDkuNWg4IiAvPgo8L3N2Zz4=.svg"/><br/>Open lucide studio</a>

ericfennis reviewed (changes requested):
Thanks for your design!
It would be nice if it matches better with the other [`list`](https://lucide.dev/icons/?search=list) icons.
<img width="426" height="425" alt="image" src="https://github.com/user-attachments/assets/f8c3b67a-8a03-4503-85f9-14469ecbc820" />

Also, name this to `list-chevrons-down-up` to make it more general.

juliankellydesign commented:
Hi Eric, I've updated to better match the other list icons and changed the name as requested. The new icon is here in this PR: https://github.com/lucide-icons/lucide/pull/3492
